actioncable/lib/action_cable/connection/subscriptions.rb

```diff
@@ -29,7 +29,13 @@ def add(data)
         subscription_klass = connection.server.channel_classes[id_options[:channel]]
 
         if subscription_klass
-          subscriptions[id_key] ||= subscription_klass.new(connection, id_key, id_options)
+          if (existing_subscription = subscriptions[id_key])
+            message = "Subscription on #{id_options[:channel]} already exists for"
+            logger.error "#{message}: #{id_key}"
+            existing_subscription.transmit_client_error({ message: "#{message} the given identifier" })
```

Comment:
Human readable error messages are all right, but it might be better to give a kind instead. E.g.:

```ruby
existing_subscription.transmit_client_error(type: :subscribing_to_existing_subscription)
```

This would better let us document different kinds of errors a client might receive.